### PR TITLE
Improve IM Message encoding

### PR DESCRIPTION
**Pull request by yunhanw-google**
#### Problem
-- Remove redundant return.
-- Add error code, CHIP_ERROR_IM_MALFORMED_EVENT_FILTER_IB
CHIP_ERROR_IM_MALFORMED_EVENT_STATUS_IB
CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_REPORT_IB
CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_STATUS_IB
CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_IB
-- Remove unused CHIP_ERROR_IM_MALFORMED_WRITE_RESPONSE_MESSAGE
-- Use enum class for fields in CommandPathIB
-- Loose the schema check for missing flag in CommandPathIB, StatusResponseMessage, similar to other IB and messages for forward compatibility purpose(https://github.com/project-chip/connectedhomeip/issues/18440)
-- Replace one exit style with ReturnErrorOnFailure
-- Remove IsProxy flag from subscription request message per latest spec change.
-- Rename kData to kFields in CommandDataIB per latest spec change

#### Change overview
See above

#### Testing
Existing tests cover

### CODE REVIEW

**Review by yufengwangca — APPROVED** [review]

**Review by mrjerryjohns — APPROVED** [review]

File: src/lib/core/CHIPError.cpp [hunk 1]
```diff
@@ -588,21 +588,21 @@ bool FormatCHIPError(char * buf, uint16_t bufSize, CHIP_ERROR err)
...
-    case CHIP_ERROR_IM_MALFORMED_COMMAND_PATH.AsInteger():
-        desc = "Malformed Interacton Model Command Path";
+    case CHIP_ERROR_IM_MALFORMED_COMMAND_PATH_IB.AsInteger():
+        desc = "Malformed Interacton Model Command Path IB";
         break;
     case CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_STATUS_ELEMENT.AsInteger():
         desc = "Malformed Interacton Model Attribute DataElement";
         break;
     case CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT.AsInteger():
         desc = "Malformed Interacton Model Attribute DataElement";
```
Comment (bzbarsky-apple) [inline src/lib/core/CHIPError.cpp:603]: What is a DataElement?

Applies to some other strings around here too.

File: src/app/MessageDef/AttributeStatusIB.cpp [hunk 2]
```diff
@@ -92,28 +92,29 @@ CHIP_ERROR AttributeStatusIB::Parser::CheckSchemaValidity() const
         {
             err = CHIP_NO_ERROR;
         }
+        else
+        {
+            err = CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_STATUS_IB;
```
[the remark this answers is not in the document]
Comment (yunhanw-google) [reply src/app/MessageDef/AttributeStatusIB.cpp]: updated across. thanks

File: src/lib/core/CHIPError.cpp (hunk 1 above)
Comment (yunhanw-google) [reply src/lib/core/CHIPError.cpp]: update with Data IB, fix across as well, thanks

File: src/lib/core/CHIPError.h [hunk 3]
```diff
@@ -2410,6 +2410,42 @@ using CHIP_ERROR = ::chip::ChipError;
...
 #define CHIP_ERROR_IM_MALFORMED_STATUS_RESPONSE_MESSAGE      CHIP_CORE_ERROR(0xe1)
 
+/**
+ * @def CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_REPORT_IB
+ *
+ * @brief
+ *   The AttributeReportIB is malformed: it either does not contain
+ *   the required elements
+ */
+#define CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_REPORT_IB             CHIP_CORE_ERROR(0xe2)
```
[the remark this answers is not in the document]
Comment (yunhanw-google) [reply src/lib/core/CHIPError.h]: done, thanks

File: src/lib/core/CHIPError.h [hunk 4]
```diff
@@ -2410,6 +2410,42 @@ using CHIP_ERROR = ::chip::ChipError;
...
+ *   the required elements
+ */
+#define CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_IB             CHIP_CORE_ERROR(0xe4)
+
+/**
+ * @def CHIP_ERROR_IM_MALFORMED_EVENT_STATUS_IB
+ *
+ * @brief
+ *   The Event Status IB is malformed: it either does not contain
+ *   the required elements
```
[the remark this answers is not in the document]
Comment (yunhanw-google) [reply src/lib/core/CHIPError.h]: removed, thanks

File: src/lib/core/CHIPError.h [hunk 5]
```diff
@@ -2410,6 +2410,42 @@ using CHIP_ERROR = ::chip::ChipError;
...
+ */
+#define CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_REPORT_IB             CHIP_CORE_ERROR(0xe2)
+
+/**
+ * @def CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_REPORT_IB
+ *
+ * @brief
+ *   The AttributeStatusIB is malformed: it either does not contain
+ *   the required elements
```
[the remark this answers is not in the document]
Comment (yunhanw-google) [reply src/lib/core/CHIPError.h]: removed, thanks

File: src/lib/core/CHIPError.h [hunk 6]
```diff
@@ -2410,6 +2410,42 @@ using CHIP_ERROR = ::chip::ChipError;
  */
 #define CHIP_ERROR_IM_MALFORMED_STATUS_RESPONSE_MESSAGE      CHIP_CORE_ERROR(0xe1)
 
+/**
+ * @def CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_REPORT_IB
+ *
+ * @brief
+ *   The AttributeReportIB is malformed: it either does not contain
+ *   the required elements
```
[the remark this answers is not in the document]
Comment (yunhanw-google) [reply src/lib/core/CHIPError.h]: removed, thanks

File: src/lib/core/CHIPError.cpp [hunk 7]
```diff
@@ -515,7 +527,7 @@ bool FormatCHIPError(char * buf, uint16_t bufSize, CHIP_ERROR err)
     case CHIP_ERROR_TOO_MANY_SHARED_SESSION_END_NODES.AsInteger():
         desc = "Too many shared session end nodes";
         break;
-    case CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_DATA_ELEMENT.AsInteger():
+    case CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_DATA_IB.AsInteger():
         desc = "Malformed Interaction Model Attribute DataElement";
```
Comment (bzbarsky-apple) [inline src/lib/core/CHIPError.cpp:531]: Again, what is a DataElement (in the description string)?

File: src/lib/core/CHIPError.cpp [hunk 8]
```diff
@@ -587,25 +599,22 @@ bool FormatCHIPError(char * buf, uint16_t bufSize, CHIP_ERROR err)
...
+        desc = "Malformed Interacton Model Command Path IB";
         break;
-    case CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT.AsInteger():
-        desc = "Malformed Interacton Model Attribute DataElement";
+    case CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_STATUS_IB.AsInteger():
+        desc = "Malformed Interacton Model Attribute Status IB";
         break;
-    case CHIP_ERROR_IM_MALFORMED_EVENT_DATA_ELEMENT.AsInteger():
+    case CHIP_ERROR_IM_MALFORMED_EVENT_DATA_IB.AsInteger():
         desc = "Malformed Interacton Model Event DataElement";
```
Comment (bzbarsky-apple) [inline src/lib/core/CHIPError.cpp:615]: What is a DataElement?

File: src/lib/core/CHIPError.h [hunk 9]
```diff
@@ -1527,12 +1527,32 @@ using CHIP_ERROR = ::chip::ChipError;
...
+#define CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_REPORT_IB             CHIP_CORE_ERROR(0x7a)
 
-// unused CHIP_CORE_ERROR(0x7a)
-
-// unused CHIP_CORE_ERROR(0x7b)
+/**
+ * @def CHIP_ERROR_IM_MALFORMED_EVENT_STATUS_IB
+ *
+ * @brief
+ *   The Event Status IB is malformed: itf does not contain
```
Comment (bzbarsky-apple) [inline src/lib/core/CHIPError.h:1543]:
```suggestion
 *   The Event Status IB is malformed: it does not contain
```

File: src/lib/core/CHIPError.cpp (hunk 7 above)
Comment (yunhanw-google) [reply src/lib/core/CHIPError.cpp]: good catch, just review all these description, and correct them, thanks

File: src/lib/core/CHIPError.cpp (hunk 8 above)
Comment (yunhanw-google) [reply src/lib/core/CHIPError.cpp]: good catch, just review all these description, and correct them, thanks

File: src/lib/core/CHIPError.h (hunk 9 above)
Comment (yunhanw-google) [reply src/lib/core/CHIPError.h]: done, thanks

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**